services/web/server/src/simcore_service_webserver/projects/project_lock.py

```diff
@@ -1,64 +1,83 @@
+from contextlib import asynccontextmanager, suppress
 from typing import Optional, Union
 
-import aioredlock
+import aioredis.exceptions
+import aioredis.lock
 from aiohttp import web
 from models_library.projects import ProjectID
 from models_library.projects_state import Owner, ProjectLocked, ProjectStatus
 
-from ..redis import get_redis_lock_manager, get_redis_lock_manager_client
+from ..redis import get_redis_lock_manager_client
 from ..users_api import UserNameDict
 
-PROJECT_REDIS_LOCK_KEY: str = "project:{}"
+PROJECT_REDIS_LOCK_KEY: str = "project_lock:{}"
 
-ProjectLock = aioredlock.Lock
-ProjectLockError = aioredlock.LockError
+ProjectLock = aioredis.lock.Lock
+ProjectLockError = aioredis.exceptions.LockError
 
 
+@asynccontextmanager
 async def lock_project(
     app: web.Application,
     project_uuid: Union[str, ProjectID],
     status: ProjectStatus,
     user_id: int,
     user_name: UserNameDict,
-) -> ProjectLock:
+):
     """returns a distributed redis lock on the project defined by its UUID.
     NOTE: can be used as a context manager
 
     try:
-        async with await lock_project(app, project_uuid, ProjectStatus.CLOSING, user_id, user_name):
+        async with lock_project(app, project_uuid, ProjectStatus.CLOSING, user_id, user_name):
             close_project(project_uuid) # do something with the project that requires the project to be locked
 
 
-    except aioredlock.LockError:
+    except ProjectLockError:
         pass # the lock could not be acquired
 
     """
-    return await get_redis_lock_manager(app).lock(
-        PROJECT_REDIS_LOCK_KEY.format(project_uuid),
-        lock_timeout=None,
-        lock_identifier=ProjectLocked(
-            value=True,
-            owner=Owner(user_id=user_id, **user_name),
-            status=status,
-        ).json(),
+    redis_lock = get_redis_lock_manager_client(app).lock(
+        PROJECT_REDIS_LOCK_KEY.format(project_uuid)
     )
+    try:
+        if not await redis_lock.acquire(
+            blocking=False,
```

Comment:
Guess you already verified, but is this the same pattern used but the previous lock? Did it not block when acquiring?

Reply:
good catch... I need to check with @pcrespov cause that part of the code is actually very weird and suspicious